src/asynqp/routing.py

```diff
@@ -21,8 +18,15 @@ def remove_handler(self, channel_id):
     def dispatch(self, frame):
         if isinstance(frame, frames.HeartbeatFrame):
             return
-        handler = self.handlers[frame.channel_id]
-        handler(frame)
+        # XXX: Could not find a better way to close channels...
+        elif isinstance(frame.payload, (spec.ConnectionCloseOK,
+                                        spec.ConnectionClose)):
+            # Notify all connection channels about it
+            for handler in self.handlers.values():
+                handler(frame)
```

Comment:
It looks like the idea is to always dispatch `ConnectionClose` to all handlers. Does the existing 'poison pill' infrastructure not serve your purpose?